book/modules/creating_modules.md

```diff
@@ -0,0 +1,473 @@
+
+::: note
+Technically, you can import this either using the directory form above or explicitly with `use increment/mod.nu *`, but the directory shorthand is preferred when using a `mod.nu`.
+:::
+
+## Subcommands
+
+As covered in [Custom Commands](../custom_commands.md), subcommands allow us to group commands logically. Using modules, this can be done in one of two ways:
+
+1. As with any custom command, the command can be defined as `"<command> <subcommand>"`, using a space inside quotes. Let's add an `increment by` subcommand to the `increment` module we defined above:
+
+```nu
+export def main []: int -> int {
+    $in + 1
+}
+
+export def "increment by" [amount: int]: int -> int {
+  $in + $amount
+}
+```
+
+It can then be imported with `use increment *` to load both the `increment` command and `increment by` subcommand.
+
+2. Alternatively, we can define the subcommand simply using the name `by`, since importing the entire `increment` module will result in the same commands:
+
+```nu
+export def main []: int -> int {
+    $in + 1
+}
+
+export def by [amount: int]: int -> int {
+  $in + $amount
+}
+```
+
+This module is imported using `use increment` (without the glob `*`) and results in the same `increment` command and `increment by` subcommand.
+
+::: note
+We'll continue to use this version for further examples below, so notice that the import pattern has changed to `use increment` (rather than `use increment *`) below.
+:::
+
+## Submodules
+
+Submodules are modules that are exported from another module. There are two ways to add a submodule to a module:
+
+1. With `export module`: Exports (a) the submodule and (b) its definitions as members of the submodule
+2. With `export use`: Exports (a) the submodule and (b) its definitions as members of the parent module
+
+To demonstrate the difference, let's add to our `increment` example by:
+
+- Adding a new `range-into-list` module and command
+- Creating a new parent module `my-utils` with `increment` and `range-into-list` as its submodules
+
+1. Create a directory for the new `my-utils` and move the `increment.nu` into it
+
+   ```nu
+   mkdir my-utils
+   # Adjust the following as needed
+   mv increment/mod.nu my-utils/increment.nu
+   rm increment
+   cd my-utils
+   ```
+
+2. In the `my-utils` directory, create a `range-into-list.nu` file with the following:
+
+   ```nu
+   export def main []: range -> list {
+       # It looks odd, yes, but the following is just
+       # a simple way to convert ranges to lists
+       each {||}
+   }
+   ```
+
+3. Test it:
+
+   ```nu
+   use range-into-list.nu
+   1..5 | range-into-list | describe
+   # => list<int> (stream)
+   ```
+
+4. We should now have a `my-utils` directory with the:
+
+   - `increment.nu` module
+   - `range-into-list.nu` module
+
+The following examples show how to create a module with submodules.
+
+### Example: Submodule with `export module`
```

Comment:
I don't think these examples makes it clear when to use `export module` and when to use `export use`, since the examples have nearly the same behavior. I don't have an example off the top of my head on how to demonstrate this, perhaps it could be demonstrated with constants?

Reply:
Good point - And there's really only one difference that I know of, and it's a doozy of a corner-case.  I've added it to the "Additional Examples" with a reference from this section.